packages/tools/src/stateManagement/segmentation/polySeg/Labelmap/convertContourToLabelmap.ts

```diff
@@ -0,0 +1,285 @@
+import {
+  StackViewport,
+  Types,
+  VolumeViewport,
+  cache,
+  utilities,
+  getWebWorkerManager,
+  volumeLoader,
+  imageLoader,
+  metaData,
+  Enums,
+  triggerEvent,
+  eventTarget,
+} from '@cornerstonejs/core';
+import { Annotation, ContourSegmentationData } from '../../../../types';
+import { getAnnotation } from '../../..';
+import { vec3 } from 'gl-matrix';
+import { Events } from '../../../../enums';
+
+const workerManager = getWebWorkerManager();
+
+export async function convertContourToVolumeLabelmap(
+  contourRepresentationData: ContourSegmentationData,
+  options: {
+    segmentIndices?: number[];
+    segmentationRepresentationUID?: string;
+    viewport?: Types.IViewport;
+  } = {}
+) {
+  const { viewport } = options;
+
+  let imageIds;
+  if (viewport instanceof VolumeViewport) {
+    const defaultActor = viewport.getDefaultActor();
+    const volumeId = defaultActor.uid;
+    const volume = cache.getVolume(volumeId);
+    imageIds = volume.imageIds;
+  } else if (viewport instanceof StackViewport) {
+    imageIds = viewport.getImageIds();
+  }
+
+  if (!imageIds) {
+    throw new Error(
+      'No imageIds found, labelmap computation from contour requires viewports with imageIds'
+    );
+  }
+
+  const segmentationVolumeId = utilities.uuidv4();
+
+  const volumeProps = utilities.generateVolumePropsFromImageIds(
+    imageIds,
+    segmentationVolumeId
+  );
+
+  const { metadata, dimensions, origin, direction, spacing, scalarData } =
+    volumeProps;
+
+  const segmentationVolume = await volumeLoader.createLocalSegmentationVolume(
+    {
+      dimensions,
+      origin,
+      direction,
+      spacing,
+      metadata,
+      imageIds: imageIds.map((imageId) => `generated://${imageId}`),
+      referencedImageIds: imageIds,
+    },
+    segmentationVolumeId
+  );
+
+  const { segmentIndices, annotationUIDsInSegmentMap } =
+    _getAnnotationMapFromSegmentation(contourRepresentationData, options);
+
+  triggerEvent(eventTarget, Events.POLYSEG_CONVERSION_STARTED, {});
+
+  const newScalarData = await workerManager.executeTask(
+    'polySeg',
+    'convertContourToVolumeLabelmap',
+    {
+      segmentIndices,
+      dimensions,
+      scalarData,
+      origin,
+      direction,
+      spacing,
+      annotationUIDsInSegmentMap,
+    },
+    {
+      callbacks: [
+        (progress) => {
```

Comment:
Consider adding a progress trigger event - maybe with a callback to cancel it?  That would then call events with increasing progress count, and would all use a single event type.